#cloud-config
repo_update: true
output:
  init:
    output: "> /var/log/cloud-init.out"
    error: "> /var/log/cloud-init.err"
  config: "tee -a /var/log/cloud-config.log"
  final:
    - ">> /var/log/cloud-final.out"
    - "/var/log/cloud-final.err"
packages:
- nginx
- neofetch
- git 
- httpie 
- vim 
- jq 
- zip 
- unzip
- tmux
- nmap
- curl
- wget
- git
- vim
write_files:
- path: /home/ubuntu/docker-compose.yaml
  content: |
    version: "3.8"
    services:
      project-zomboid:
        container_name: pzserver
        image: pepecitron/projectzomboid-server
        restart: unless-stopped
        environment:
          SERVER_ADMIN_PASSWORD: "pzadmin"
          SERVER_PASSWORD: "secretpassword"
        ports:
          - "8766:8766/udp"
          - "8767:8767/udp"
          - "16261:16261/udp"
        volumes:
          - ./data/server-file:/data/server-file
          - ./data/config:/data/config
runcmd:
- systemctl start nginx
- systemctl enable nginx
- [curl, -fsSL, https://get.docker.com, -o, get-docker.sh]
- [sh, get-docker.sh]
- service docker start
- [usermod, -aG, docker, $USER]
- [usermod, -aG, docker, ubuntu]
- cd /home/ubuntu/
- [git, clone,https://github.com/PepeCitron/projectzomboid-server.git]
- docker compose up -d 

final_message: "The system is finally up, after $UPTIME seconds"